Kytos unit test failing: tests/unit/test_core/test_controller.py::TestController::test_create_pidfile

Hi Kytos team,

I'm running the unit tests from Kytos using gitlab-ci in the following way:
```
  image: python:3.6
  script:
    - apt-get update -y && apt-get upgrade -y
    - git config --global url."https://github.com".insteadOf git://github.com
    - pip install --upgrade pip tox==3.14.6
    - git clone https://github.com/kytos/kytos
    - cd kytos
    - tox
```

The unit tests are failing with the following error:
```
=================================== FAILURES ===================================
203______________________ TestController.test_create_pidfile ______________________
204self = <kytos.core.controller.Controller object at 0x7f101db64630>
205    def create_pidfile(self):
206        """Create a pidfile."""
207        pid = os.getpid()
208    
209        # Creates directory if it doesn't exist
210        # System can erase /var/run's content
211        pid_folder = Path(self.options.pidfile).parent
212        self.log.info(pid_folder)
213    
214        # Pylint incorrectly infers Path objects
215        # https://github.com/PyCQA/pylint/issues/224
216        # pylint: disable=no-member
217        if not pid_folder.exists():
218            pid_folder.mkdir()
219            pid_folder.chmod(0o1777)
220        # pylint: enable=no-member
221    
222        # Make sure the file is deleted when controller stops
223        atexit.register(Path(self.options.pidfile).unlink)
224    
225        # Checks if a pidfile exists. Creates a new file.
226        try:
227>           pidfile = open(self.options.pidfile, mode='x')
228E           FileExistsError: [Errno 17] File exists: '/tmp/tmpfdharbdn'
229kytos/core/controller.py:214: FileExistsError
230During handling of the above exception, another exception occurred:
231self = <tests.unit.test_core.test_controller.TestController testMethod=test_create_pidfile>
232args = (<MagicMock name='atexit' id='139707194440280'>, <MagicMock name='getpid' id='139707194526128'>)
233_ = <MagicMock name='atexit' id='139707194440280'>
234mock_getpid = <MagicMock name='getpid' id='139707194526128'>
235tmp_file = <tempfile._TemporaryFileWrapper object at 0x7f101db7e2e8>
236    @patch('os.getpid')
237    @patch('kytos.core.controller.atexit')
238    def test_create_pidfile(self, *args):
239        """Test activate method."""
240        (_, mock_getpid) = args
241        mock_getpid.return_value = 2
242        with tempfile.NamedTemporaryFile() as tmp_file:
243            tmp_file.write(b'1')
244            tmp_file.seek(0)
245            self.controller.options.pidfile = tmp_file.name
246    
247>           self.controller.create_pidfile()
248tests/unit/test_core/test_controller.py:190: 
249_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
250self = <kytos.core.controller.Controller object at 0x7f101db64630>
251    def create_pidfile(self):
252        """Create a pidfile."""
253        pid = os.getpid()
254    
255        # Creates directory if it doesn't exist
256        # System can erase /var/run's content
257        pid_folder = Path(self.options.pidfile).parent
258        self.log.info(pid_folder)
259    
260        # Pylint incorrectly infers Path objects
261        # https://github.com/PyCQA/pylint/issues/224
262        # pylint: disable=no-member
263        if not pid_folder.exists():
264            pid_folder.mkdir()
265            pid_folder.chmod(0o1777)
266        # pylint: enable=no-member
267    
268        # Make sure the file is deleted when controller stops
269        atexit.register(Path(self.options.pidfile).unlink)
270    
271        # Checks if a pidfile exists. Creates a new file.
272        try:
273            pidfile = open(self.options.pidfile, mode='x')
274        except OSError:
275            # This happens if there is a pidfile already.
276            # We shall check if the process that created the pidfile is still
277            # running.
278            try:
279                existing_file = open(self.options.pidfile, mode='r')
280                old_pid = int(existing_file.read())
281                os.kill(old_pid, 0)
282                # If kill() doesn't return an error, there's a process running
283                # with the same PID. We assume it is Kytos and quit.
284                # Otherwise, overwrite the file and proceed.
285                error_msg = ("PID file {} exists. Delete it if Kytos is not "
286                             "running. Aborting.")
287>               sys.exit(error_msg.format(self.options.pidfile))
288E               SystemExit: PID file /tmp/tmpfdharbdn exists. Delete it if Kytos is not running. Aborting.
289kytos/core/controller.py:228: SystemExit
290
```

--- a/tests/unit/test_core/test_controller.py
+++ b/tests/unit/test_core/test_controller.py
@@ -183,7 +183,7 @@ class TestController(TestCase):
         (_, mock_getpid) = args
         mock_getpid.return_value = 2
         with tempfile.NamedTemporaryFile() as tmp_file:
-            tmp_file.write(b'4194305') # pid_max +1
+            tmp_file.write(b'4194305')  # pid_max +1
             tmp_file.seek(0)
             self.controller.options.pidfile = tmp_file.name
 

--- a/tests/unit/test_core/test_controller.py
+++ b/tests/unit/test_core/test_controller.py
@@ -183,7 +183,7 @@ class TestController(TestCase):
         (_, mock_getpid) = args
         mock_getpid.return_value = 2
         with tempfile.NamedTemporaryFile() as tmp_file:
-            tmp_file.write(b'1')
+            tmp_file.write(b'4194305') # pid_max +1
             tmp_file.seek(0)
             self.controller.options.pidfile = tmp_file.name
 

